inference-engine/tests/functional/plugin/cpu/test_utils/fusing_test_utils.hpp

```diff
@@ -0,0 +1,138 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "cpu_test_utils.hpp"
+#include <shared_test_classes/single_layer/activation.hpp>
+
+namespace CPUTestUtils {
+
+struct postNode {
+    std::function<std::shared_ptr<ngraph::Node>(std::shared_ptr<ngraph::Node>, const ngraph::element::Type&, ngraph::ParameterVector&)> makeNode;
+    std::string name;
+};
+
+typedef std::tuple<
+        std::shared_ptr<ngraph::Function>,
+        std::vector<postNode>,
+        std::vector<std::string>> fusingSpecificParams;
+
+class CpuTestWithFusing : public CPUTestsBase {
+public:
+    static std::string getTestCaseName(fusingSpecificParams params);
+
+protected:
+    std::shared_ptr<ngraph::Node> modifyGraph(const ngraph::element::Type &ngPrc,
+                                                      ngraph::ParameterVector &params,
+                                                      const std::shared_ptr<ngraph::Node> &lastNode) const override;
+
+    void CheckPluginRelatedResults(InferenceEngine::ExecutableNetwork &execNet, std::string nodeType) const override;
+
+private:
+    std::shared_ptr<ngraph::Node> addPostFunction(const std::shared_ptr<ngraph::Node> &lastNode) const;
+    std::shared_ptr<ngraph::Node> addPostNodes(const ngraph::element::Type &ngPrc,
+                                               ngraph::ParameterVector &params,
+                                               const std::shared_ptr<ngraph::Node> &lastNode) const;
+protected:
+    std::shared_ptr<ngraph::Function> postFunction;
+    std::vector<postNode> postNodes;
+    std::vector<std::string> fusedOps;
+};
+
+/* FUSING PATTERNS */
+const auto emptyFusingSpec = fusingSpecificParams{nullptr, {}, {}};
+const auto fusingRelu = fusingSpecificParams{nullptr, {
+            {[](std::shared_ptr<ngraph::Node> inpNode, const ngraph::element::Type& ngPrc, ngraph::ParameterVector& params){
```

Comment:
Precision as a part of the builder interface could be useful in general case since it gives little bit more flexibility. For example if we want to sum with parameter of a different type.

Reply:
In that case it is better to pass vector of precisions to cover when we have several inputs in builder. However lets refactor this once we really need it.